File: pt_example.js
/*
  Usage:
  node pt_example.js pt_cities.txt
*/

// Make sure we got a filename on the command line.
if (process.argv.length < 3) {
    console.log('Usage: node ' + process.argv[1] + ' FILENAME');
    process.exit(1);
}
// Read the file
var fs = require('fs'),
    filename = process.argv[2];
fs.readFile(filename, 'utf8', function(err, data) {
    if (err) throw err;
    // replace _ with space
    var txt = data.replace(/_/g, " ");
    var lines = txt.split('\n');
    //remove header
    lines = lines.splice(1);
    lines.forEach(function(line) {
        var fields = line.split('\t');
        // fields|id name lat lon countryCode
        console.log("----------");
        console.log("Cidade: ");
        console.log("id: ", fields[0]);
        console.log("nome: ", fields[1]);
        console.log("latitude: ", fields[2]);
        console.log("longitude: ", fields[3]);
        console.log("código de país: ", fields[4]);
    });

});

File: pt_example.cs
using System;
using System.Globalization;
using System.IO;

namespace citiesList
{
    class pt_example
    {
        static void Main(string[] args)
        {
            string[] lines = File.ReadAllLines("pt_cities.txt");
            int i = 1; // Line number 
            foreach (var line in lines)
            {
                string[] tokens = line.Split(new char[] { '\t' });

                long id = long.Parse(tokens[0].ToString());
                string name = tokens[1].ToString().Replace("_", " "); // Replace underscore by space
                float latitude = float.Parse(tokens[2].ToString(), CultureInfo.InvariantCulture); // InvariantCulture: handles FormatException
                float longitude = float.Parse(tokens[3].ToString(), CultureInfo.InvariantCulture);
                string countryCode = tokens[4].ToString();

                // Print cities
                Console.WriteLine("-- #" + i);
                Console.WriteLine("ID: " + id);
                Console.WriteLine("Name: " + name);
                Console.WriteLine("Latitude: " + latitude);
                Console.WriteLine("Longitude: " + longitude);
                Console.WriteLine("Country Code: " + countryCode);

                i++;
            }
        }
    }
}
